FishyTransport/Sources/FishyActorsCore/SourceGen.swift

```diff
@@ -0,0 +1,239 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the fishy-actor-transport open source project
+//
+// Copyright (c) 2021 Apple Inc. and the fishy-actor-transport project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of fishy-actor-transport project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+import Foundation
+
+public final class SourceGen {
+  static let header = String(
+    """
+    // DO NOT MODIFY: This file will be re-generated automatically.
+    // Source generated by FishyActorsGenerator (version x.y.z)
+    import _Distributed
+    
+    import FishyActorTransport
+    import ArgumentParser
+    import Logging
+    
+    import func Foundation.sleep
+    import struct Foundation.Data
+    import class Foundation.JSONDecoder
+    
+    
+    """)
+
+  var buckets: Int
+
+  public init(buckets: Int) {
+    if buckets > 1 { print("Warning: requested \(buckets) buckets, but bucketing is not implemented yet; defaulting to 1 bucket.") }
+    self.buckets = 1 // TODO: hardcoded for now, would use bucketing approach to avoid re-generating too many sources
+
+    // TODO: Don't do this in init
+    // Just make sure all "buckets" exist
+//    for i in (0..<buckets) {
+//      let path = targetFilePath(targetDirectory: targetDirectory, i: i)
+//      try! SourceGen.header.write(to: path, atomically: true, encoding: .utf8)
+//    }
+  }
+
+  public func generate(decl: DistributedActorDecl) -> [String] {
+    return [try! generateSources(for: decl)]
```

Comment:
Seems like this is a good place to actually keep throwing, wdyt?
```suggestion
    return try [generateSources(for: decl)]
```

Reply:
The signature seems wrong?

The gen must decide into which bucket this generated source goes and it is always exactly one source text. (well, in this impl at least).

The result should be `GeneratedSource { let text: String; let bucket: Int }`.